### PR TITLE
Use Green Bank Convention to write and read vectorized location for time coordinates to/from FITS

### PR DESCRIPTION
**Pull request by aaryapatil**
This PR allows the user to write and read vectorized location for time coordinates following the ``astropy.time.Time`` machinery. Refer to the [FITS Green Bank](https://fits.gsfc.nasa.gov/registry/greenbank.html) convention for details.

@mhvk @taldcroft We had decided to do this as it is fairly simple. I have added some basic tests.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -164,6 +164,9 @@ astropy.stats
 astropy.table
 ^^^^^^^^^^^^^
 
+- Added a feature to write and read vectorized location for time coordinates
```
Comment (bsipocz) [inline CHANGES.rst]: @taldcroft - is this an io.fits or table feature?

**Comment by mhvk** [comment]
Wow, that is an impressively *small* change for quite a big feature! Will try to look more closely soon.

**Comment by pllim** [comment]
Does this need a "what's new"?

File: astropy/io/fits/tests/test_fitstime.py [hunk 2]
```diff
@@ -41,12 +41,20 @@ def test_time_to_fits_loc(self, table_types):
         t['a'] = Time(self.time, format='isot', scale='utc')
```
[the remark this answers is not in the document]
Comment (aaryapatil) [reply astropy/io/fits/tests/test_fitstime.py]: Have taken care of this!

File: astropy/io/fits/fitstime.py [hunk 3]
```diff
@@ -550,14 +555,17 @@ def time_to_fits(table):
...
                 # Set global geocentric location
                 location = col.location
-                hdr.extend([Card(keyword='OBSGEO-{}'.format(dim.upper()),
-                                 value=getattr(location, dim).to_value(u.m))
+                if col.location.size > 1:
+                    for dim in ('x', 'y', 'z'):
+                        newtable.add_column(Column(getattr(location, dim).to_value(u.m)),
+                                            name='OBSGEO-{}'.format(dim.upper()))
+                else:
+                    hdr.extend([Card(keyword='OBSGEO-{}'.format(dim.upper()),
```
[the remark this answers is not in the document]
Comment (aaryapatil) [reply astropy/io/fits/fitstime.py]: Did changes to ``Time.__init__``.

**Comment by aaryapatil** [comment]
Hopefully, all tests are done! (phew) Now all that is remaining is documentation update. I'll get to that soon.

**Comment by aaryapatil** [comment]
@mhvk I have updated the documentation. I think this is ready for a final review. Let me know whether a separate documentation is required which explains the Green Bank convention. I think adding a link to the convention should be enough (I have done so).

**Comment by mhvk** [comment]
One request: could you rebase/squash to combine commits a little (getting rid of the "silly mistake" ones...).

Also, just right the extent to which you describe the greenbank convention.

**Comment by aaryapatil** [comment]
@mhvk I've squashed the commits.

**Comment by mhvk** [comment]
great, thanks, @aaryapatil!

**Pull request merged**